docs/source/manual/file_formats/pcf_custom_command_config_file.rst

```diff
@@ -0,0 +1,117 @@
+.. _file_format_pcf_custom_command_config_file:
+
+PCF Custom Command Configuration File (.xml)
+--------------------------------------------
+
+.. note:: This file defines custom PCF commands that can be used in :ref:`file_format_pin_constraints_file`.  
+   It specifies how user-facing commands are translated into FPGA configuration bits. See details in :ref:`openfpga_setup_commands_pcf2bitstream_setting`.
+
+PCF is the unified file used by users to manipulate FPGA I/Os. Custom commands in PCF provide a simple and straightforward way for users to configure FPGA I/Os into their desired operating modes without understanding the underlying bitstream-level details. The custom command configuration file encapsulates the mapping from these high-level commands to the corresponding bit-level configuration details.
+
+
+The following diagram illustrates how the PCF custom commands configure FPGA modules through the management stack and special I/O interface:
+
+
+.. _fig_pcf_custom_command:
+
+.. figure:: ./figures/pcf_custom_command.png
+   :width: 100%
+   :alt: An example of FPGA Domain with Management Stack and Fabric 
+
+   An example of FPGA Domain with Management Stack and Fabric
+
+As shown in the figure :numref:`fig_pcf_custom_command`
+
+- **FPGA Management Stack** contains functional modules such as the Delay Chain and WDT.
+  Each module receives configuration bits from the PCF parser based on the user-specified
+  commands.
+- **Special I/O** interfaces transfer configuration bits from the management stack to
+  the corresponding hardware blocks in the **FPGA Fabric**.
+- Users interact only with high-level commands in the PCF file. For example, to configure
+  a programmable delay chain in an I/O, the user simply specifies:
+
+::
+
+  set_delay_chain -pad pad_io[0] -delay 0.2
+
+where ``set_delay_chain`` is defined in the following PCF custom command configuration file:
+
+.. code-block:: xml
+
+  <pcf_config>
+    <command name="set_delay_chain" type="delay_chain">
+      <option name="pad" type="pin"/>
+      <pb_type name="gp_inpad.inpad"/>
+      <option name="delay" type="mode" offset="0">
+        <mode name="0.1ns">00001</mode>
+        <mode name="0.2ns">00010</mode>
+        <mode name="0.3ns">00100</mode>
+      </option>
+    </command>
+
+    <command name="set_watch_dog" type="peripheral">
+      <option name="pad" type="pin"/>
+      <pb_type name="gp_outpad.outpad"/>
+      <option name="mode" type="decimal" num_bits="3" max="6" little_endian="false" offset="0"/>
+    </command>
+  </pcf_config>
+
+The parser converts value ``0.2`` to the corresponding bit pattern ``00010`` and programs the Delay Chain via the Special I/O.
```

Comment:
This is not expected. You define what you will get. We should not prune the mode name. Please double check.